src/common/components/repository-row/index.js

```diff
@@ -154,6 +154,17 @@ class RepositoryRow extends Component {
         this.props.dispatch(registerNameClear(this.props.fullName));
       }, 2000);
     }
+
+    // Close snap name mismatch dropdown if
+    // name is updated and fixed
+    const oldSnapProp = this.props.snap;
+    const newSnapProp = nextProps.snap;
+    if (snapNameIsMismatched(oldSnapProp) && !snapNameIsMismatched(newSnapProp)) {
+      this.setState({
+        ...this.state,
```

Comment:
This is not needed. `setState` will do the magic, you just need to set what you want to change.